utils/utils.go

```diff
@@ -440,36 +440,30 @@ func techArrayToString(techsArray []coreutils.Technology, separator string) (res
 
 type UrlAccessChecker struct {
 	connected *bool
-	waitGroup errgroup.Group
+	waitGroup sync.WaitGroup
```

Comment:
now that we are using wait group and we are always waiting before returning from IsConnected, we don't need a pointer for `connected` to tell us if the action is over, it can be changed to bool